lib/Agrammon/Model.pm6

```diff
@@ -389,6 +389,25 @@ class Agrammon::Model {
         [flat @singles, %multis.pairs]
     }
 
+    #| Obtains a sorted description of the structure of the model, accounting for multiple
+    #| instance modules. The list returned contains single-instance modules as
+    #| string names of their taxonomies. A multi-instance module root is a pair,
+    #| where the key is the taxonomy name of the root of the multi-instance module
+    #| and the value is an array of the modules within that instance.
+    method extract-structure-sorted($sort? = 'model') {
+        my \order = $sort eq 'calculation' ?? @!evaluation-order !! @!load-order;
+        my @modules;
+        for order -> Agrammon::Model::Module $module {
+            if $module.instance-root -> $root {
+                @modules.push( $root => $module.taxonomy );
```

Comment:
This will produce loads of pairs rather than a pair mapping to the an array of the modules. That doesn't feel terribly useful; it's creating an intermediate data structure that's not making the next step any easier...